### PR TITLE
fix: do not escape starting and ending symbols for patterns validation

### PR DESCRIPTION
**Pull request by m-lagarde**
[TR-825](https://oat-sa.atlassian.net/browse/TR-825)

The goal of this PR is to fix the pattern control from validation constraint when starting and ending symbols are provided

### CODE REVIEW

File: src/qtism/runtime/expressions/operators/Utils.php [hunk 1]
```diff
@@ -328,6 +328,8 @@ public static function lastPregErrorMessage()
      */
     public static function prepareXsdPatternForPcre($pattern)
     {
+        $pattern = trim($pattern, '^$');
+
         // XML schema always implicitly anchors the entire regular expression
         // because there is no carret (^) nor dollar ($) signs.
```
Comment (julien-sebire) [inline src/qtism/runtime/expressions/operators/Utils.php:334]: This should read:
```
// XML schema always implicitly anchors the entire regular expression.
// Neither caret (^) nor dollar ($) sign are supported as anchors.
```
It's more explicit. You can change that in the same time.

File: test/qtismtest/runtime/expressions/operators/PatternMatchProcessorTest.php [hunk 2]
```diff
@@ -110,7 +110,9 @@ public function patternMatchProvider()
             [new QtiString('string'), 'shell', false],
             [new QtiString('stringString'), '.*', true], // in xml schema 2, dot matches white-spaces
```
Comment (julien-sebire) [inline test/qtismtest/runtime/expressions/operators/PatternMatchProcessorTest.php:111]: This test doesn't actually test what the comment says. The searched string should contain at least one space:
```
[new QtiString('string String'), '.*', true], // in xml schema 2, dot matches white-spaces
```

File: test/qtismtest/runtime/expressions/operators/PatternMatchProcessorTest.php [hunk 3]
```diff
@@ -110,7 +110,9 @@ public function patternMatchProvider()
...
             [new QtiString('^String$'), 'String', false], // No carret nor dollar in xml schema 2
-            [new QtiString('^String$'), '^String$', true],
+            [new QtiString('String'), 'String', true],
+            [new QtiString('String'), '^String$', true],
+            [new QtiString('^String$'), '^String$', false],
             [new QtiString('Str/ing'), 'Str/ing', true],
```
Comment (julien-sebire) [inline test/qtismtest/runtime/expressions/operators/PatternMatchProcessorTest.php:116]: Please add tests to match `12.99$` and `$12.99`.

**Review by julien-sebire — CHANGES_REQUESTED** [review]
Nice try but, how can you match `10.99$` or `$10.99` then? :-) I'm not sure we could find use cases for matching `^` at beginning or end of a string though.
Maybe remove ^ and $ together only when they are both here?
Or give a hint in your front-end to tell that nor ^ nor $ anchors are supported and that the pattern will be matched to the whole response?
Or take them in account when escaped (`\^` and `\$`)?

File: src/qtism/runtime/expressions/operators/Utils.php [hunk 4]
```diff
@@ -328,6 +328,8 @@ public static function lastPregErrorMessage()
      */
     public static function prepareXsdPatternForPcre($pattern)
     {
+        $pattern = trim($pattern, '^$');
```
Comment (wazelin) [inline src/qtism/runtime/expressions/operators/Utils.php:331]:
```suggestion
        $pattern = rtrim(ltrim($pattern, '^'), '$');
```

**Review by wazelin — CHANGES_REQUESTED** [review]

File: test/qtismtest/runtime/expressions/operators/PatternMatchProcessorTest.php (hunk 3 above)
Comment (wazelin) [reply test/qtismtest/runtime/expressions/operators/PatternMatchProcessorTest.php:116]: Considering $ is usually prepended to a value, I would add just `$12.99` support.

**Pull request closed without merging**